src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -598,8 +691,11 @@ private bool IsUnquotedEndOfAttributeValue(SyntaxToken token)
                    token.Kind == SyntaxKind.NewLine;
         }
 
-        private void ParseJavascriptAndEndScriptTag(in SyntaxListBuilder<RazorSyntaxNode> builder, AcceptedCharactersInternal endTagAcceptedCharacters = AcceptedCharactersInternal.Any)
+        private void ParseJavascriptAndEndScriptTag(in SyntaxListBuilder<RazorSyntaxNode> builder, MarkupStartTagSyntax startTag, AcceptedCharactersInternal endTagAcceptedCharacters = AcceptedCharactersInternal.Any)
         {
+            var previousNodes = builder.ToList();
```

Comment:
By not adding the previous nodes immediately to the builder aren't we re-ordering script content and nodes prior to it?